dali/test/python/test_pipeline_debug.py

```diff
@@ -649,3 +657,128 @@ def test_nan_check():
+    def pipeline_split_merge():
+        pred = fn.random.coin_flip(seed=42, dtype=types.BOOL)
+        rng1 = fn.random.coin_flip(seed=1)
+        rng2 = fn.random.coin_flip(seed=2)
+        true, _ = fn._conditional.split(rng1, predicate=pred)
+        _, false = fn._conditional.split(rng2, predicate=pred)
+        output_true = true + 20
+        output_false = false + 10
+        output = fn._conditional.merge(output_true, output_false, predicate=pred)
+        print(
+            f"Pred: {pred}, Output if: {output_true}, Output else: {output_false}, Output {output}")
+        return pred, output
+
+    @pipeline_def(batch_size=8, num_threads=3, device_id=0, enable_conditionals=True)
+    def pipeline_cond():
+        pred = fn.random.coin_flip(seed=42, dtype=types.BOOL)
+        rng1 = fn.random.coin_flip(seed=1)
+        rng2 = fn.random.coin_flip(seed=2)
+        print(f"Pred: {pred}")
+        if pred:
+            output = rng1 + 20
+            print(f"Output if: {output}")
+        else:
+            output = rng2 + 10
+            print(f"Output else: {output}")
+        print(f"Output: {output}")
+        return pred, output
+
+    pipe_standard = pipeline_split_merge(debug=True)
+    pipe_standard.build()
+
+    pipe_cond = pipeline_cond(debug=True)
+    pipe_cond.build()
+    compare_pipelines(pipe_standard, pipe_cond, 8, 5)
+
+
+@params(*[(True, False), (False, True), (True, True)])
+def test_against_split_merge(base_debug, conditional_debug):
+    impl_test_against_split_merge({'debug': base_debug}, {'debug': conditional_debug})
+
+
+@params(*[(True, False), (False, True), (True, True)])
+def test_dot_gpu(base_debug, conditional_debug):
+    impl_test_dot_gpu({'debug': base_debug}, {'debug': conditional_debug})
+
+
+@params(*[(True, False), (False, True), (True, True)])
+def test_arg_inputs_scoped_tracking(global_debug, scoped_debug):
+    impl_test_arg_inputs_scoped_tracking({'debug': global_debug}, {'debug': scoped_debug})
+
+
+def test_arg_inputs_scoped_uninitialized():
+    impl_test_arg_inputs_scoped_uninitialized({'debug': True})
+
+
+@params(*(pred_gens[:-1]))
+def test_generators(pred):
+    for base_debug, conditional_debug in [(True, False), (False, True), (True, True)]:
+        impl_test_generators(pred, {'debug': base_debug}, {'debug': conditional_debug})
+
+
+def test_uninitialized():
+    impl_test_uninitialized({'debug': True})
+
+
+def test_debug_pipeline_conditional_without_data_node():
+    @pipeline_def(batch_size=8, num_threads=3, device_id=0, enable_conditionals=True)
+    def pipeline_cond():
+        pred = fn.random.coin_flip(seed=42, dtype=types.BOOL)
+        rng1 = fn.random.coin_flip(seed=1)
+        if pred:
+            output = fn.copy(rng1.get())
+        else:
+            output = rng1 + 10
+        return pred, output
+
+    with assert_raises(
+            ValueError, glob=("Debug mode for conditionals doesn't allow for modification of"
```

Comment:
```suggestion
            ValueError, glob=("Debug mode for conditional statements doesn't allow for modification of"
```
"conditionals" is our internal jargon.

Reply:
done